tests/integration/test_materialize_mysql_database/materialize_with_ddl.py

```
@@ -461,3 +462,52 @@ def select_without_columns(clickhouse_node, mysql_node, service_name):
+    mysql_node.query("CREATE DATABASE test_database DEFAULT CHARACTER SET 'utf8'")
+    mysql_node.query("CREATE TABLE test_database.test_table_1 (id INT NOT NULL PRIMARY KEY) ENGINE = InnoDB;")
+    mysql_node.query("INSERT INTO test_database.test_table_1 VALUES(1), (2), (3), (4), (5), (6);")
+
+    mysql_node.result("SHOW GRANTS FOR 'test'@'%';")
+
+    clickhouse_node.query(
+        "CREATE DATABASE test_database ENGINE = MaterializeMySQL('{}:3306', 'test_database', 'test', '123')".format(
+            service_name))
+
+    # wait MaterializeMySQL read binlog events
+    time.sleep(90)
+
+    assert "test_table_1" in clickhouse_node.query("SHOW TABLES FROM test_database")
+    check_query(clickhouse_node, "SELECT count() FROM test_database.test_table_1 FORMAT TSV", "6\n", 5, 5)
+    mysql_node.query("INSERT INTO test_database.test_table_1 VALUES(7);")
+    check_query(clickhouse_node, "SELECT count() FROM test_database.test_table_1 FORMAT TSV", "7\n")
+    clickhouse_node.query("DROP DATABASE test_database;")
+
+    mysql_node.query("REVOKE REPLICATION SLAVE ON *.* FROM 'test'@'%'")
+    clickhouse_node.query(
+        "CREATE DATABASE test_database ENGINE = MaterializeMySQL('{}:3306', 'test_database', 'test', '123')".format(
+            service_name))
+    assert "test_database" in clickhouse_node.query("SHOW DATABASES")
+    assert "test_table_1" not in clickhouse_node.query("SHOW TABLES FROM test_database")
+    clickhouse_node.query("DROP DATABASE test_database")
+
+    mysql_node.query("REVOKE REPLICATION CLIENT, RELOAD ON *.* FROM 'test'@'%'")
+    clickhouse_node.query(
+        "CREATE DATABASE test_database ENGINE = MaterializeMySQL('{}:3306', 'test_database', 'test', '123')".format(
+            service_name))
+    assert "test_database" in clickhouse_node.query("SHOW DATABASES")
+    assert "test_table_1" not in clickhouse_node.query("SHOW TABLES FROM test_database")
```

Maybe throw an exception on database creation?